block/manager.go

```diff
@@ -927,26 +929,33 @@ func (m *Manager) submitBlocksToDA(ctx context.Context) error {
 				}
 			}
 			m.logger.Debug("resetting DA layer submission options", "backoff", backoff, "gasPrice", gasPrice, "maxBlobSize", maxBlobSize)
+			if submittedAllBlocks {
+				break daSubmitRetryLoop
+			}
 		case da.StatusNotIncludedInBlock, da.StatusAlreadyInMempool:
 			m.logger.Error("DA layer submission failed", "error", res.Message, "attempt", attempt)
 			backoff = m.conf.DABlockTime * time.Duration(m.conf.DAMempoolTTL)
 			if m.dalc.GasMultiplier > 0 && gasPrice != -1 {
 				gasPrice = gasPrice * m.dalc.GasMultiplier
 			}
 			m.logger.Info("retrying DA layer submission with", "backoff", backoff, "gasPrice", gasPrice, "maxBlobSize", maxBlobSize)
-			time.Sleep(backoff)
-			backoff = m.exponentialBackoff(backoff)
+
 		case da.StatusTooBig:
 			m.logger.Error("DA layer submission failed", "error", res.Message, "attempt", attempt)
 			maxBlobSize = maxBlobSize / 4
-			time.Sleep(backoff)
-			backoff = m.exponentialBackoff(backoff)
+
 		default:
 			m.logger.Error("DA layer submission failed", "error", res.Message, "attempt", attempt)
-			time.Sleep(backoff)
-			backoff = m.exponentialBackoff(backoff)
+
 		}
 		attempt += 1
+
+		select {
+		case <-ctx.Done():
+			break daSubmitRetryLoop
+		case <-time.After(backoff):
+			backoff = m.exponentialBackoff(backoff)
```

Comment:
We shouldn't handle this here, because then we are exponentially backing off even on successful attempts. 
```suggestion
```